homeassistant/components/http/__init__.py

```diff
@@ -329,34 +338,97 @@ async def serve_file(request: web.Request) -> web.FileResponse:
             self.app.router.add_route("GET", url_path, serve_file)
         )
 
-    async def start(self) -> None:
-        """Start the aiohttp server."""
-        context: ssl.SSLContext | None
-        if self.ssl_certificate:
+    def _create_ssl_context(self) -> ssl.SSLContext | None:
+        context: ssl.SSLContext | None = None
+        assert self.ssl_certificate is not None
+        try:
+            if self.ssl_profile == SSL_INTERMEDIATE:
+                context = ssl_util.server_context_intermediate()
+            else:
+                context = ssl_util.server_context_modern()
+            context.load_cert_chain(self.ssl_certificate, self.ssl_key)
+        except OSError as error:
+            _LOGGER.error(
+                "Could not read SSL certificate from %s: %s",
+                self.ssl_certificate,
+                error,
+            )
+            _LOGGER.warning(
+                "The server will start up with an emergency self signed ssl certificate"
+            )
             try:
-                if self.ssl_profile == SSL_INTERMEDIATE:
-                    context = ssl_util.server_context_intermediate()
-                else:
-                    context = ssl_util.server_context_modern()
-                await self.hass.async_add_executor_job(
-                    context.load_cert_chain, self.ssl_certificate, self.ssl_key
-                )
+                return self._create_emergency_ssl_context()
```

Comment:
Let's change the flow slightly such that:

- Not being able to read SSL cert will trigger safe mode
- In Safe mode, we use a temporary cert

Reply:
<img width="869" alt="Screen Shot 2022-02-18 at 15 01 26" src="https://user-images.githubusercontent.com/663432/154760715-34407753-6b3d-4ec8-8bf5-9cd827876e6a.png">